src/platform/Linux/DeviceInfoProviderImpl.h

```diff
@@ -69,6 +81,12 @@ class DeviceInfoProviderImpl : public DeviceInfoProvider
     CHIP_ERROR SetUserLabelLength(EndpointId endpoint, size_t val) override;
     CHIP_ERROR GetUserLabelLength(EndpointId endpoint, size_t & val) override;
     CHIP_ERROR SetUserLabelAt(EndpointId endpoint, size_t index, const UserLabelType & userLabel) override;
+
+private:
+    DeviceLayer::Internal::ChipLinuxStorage mStorage;
+    DefaultStorageKeyAllocator keyAlloc;
```

Comment:
You never need an instance of this class, since it doesn't use state. Always use it on the stack at the point of use.